### PR TITLE
llms/cloudflare: Implement Cloudflare Workers AI LLM

### PR DESCRIPTION
**Pull request by rajaiswal**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

This PR implements Cloudflare Workers AI LLM as documented [here](https://developers.cloudflare.com/api/operations/workers-ai-post-run-model). Functionality is at parity with [Langhain's](https://github.com/langchain-ai/langchain) Cloudflare Workers AI [implementation](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/llms/cloudflare_workersai.py).

### CODE REVIEW

**Comment by tmc** [comment]
@rajaiswal looking forward to this! can you get this over the line and mark it ready for review?

**Comment by rajaiswal** [comment]
@tmc I just need to add tests. What's the general acceptable test pattern here? I see some llms are spinning up a mock server, and some are testing against the realy API with test tokens. Not sure which one to follow

**Comment by tmc** [comment]
The short answer is that we haven't solidified and recommended an approach yet. Would love contributions to help shape that, but in the short term do what you feel is expedient and appropriate.

**Comment by rajaiswal** [comment]
@tmc PR is ready for review. Some unrelated files got touched when I ran linting and gofumpt, would you like me to revert those files back to their original state and isolate the linting only to my changes?

File: llms/cloudflare/internal/cloudflareclient/api.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func (c Client) CreateEmbedding(ctx context.Context, texts *CreateEmbeddingRequest) (*CreateEmbeddingResponse, error) {
```
Comment (tmc) [inline llms/cloudflare/internal/cloudflareclient/api.go:14]: these receivers should use a pointer to Client to lower allocations on call.

**Review by tmc — COMMENTED** [review]
This is looking great! No worries with the unrelated format fixes. Just a few small comments here but generally this looks like it's nearly good to go in!

File: llms/cloudflare/internal/cloudflareclient/api.go (hunk 1 above)
Comment (rajaiswal) [reply llms/cloudflare/internal/cloudflareclient/api.go]: done

**Comment by rajaiswal** [comment]
@tmc this should be good to merge now

**Review by tmc — APPROVED** [review]
Thanks, LGTM

**Pull request merged**